nemo/collections/asr/models/ctc_models.py

```diff
@@ -68,22 +93,39 @@ def __setup_dataloader_from_config(config: Optional[Dict]):
     def setup_training_data(self, train_data_layer_config: Optional[Union[DictConfig, Dict]]):
         if 'shuffle' not in train_data_layer_config:
             train_data_layer_config['shuffle'] = True
-        self.__train_dl = self.__setup_dataloader_from_config(config=train_data_layer_config)
+        self._train_dl = self.__setup_dataloader_from_config(config=train_data_layer_config)
 
     def setup_validation_data(self, val_data_layer_config: Optional[Union[DictConfig, Dict]]):
         if 'shuffle' not in val_data_layer_config:
             val_data_layer_config['shuffle'] = False
-        self.__val_dl = self.__setup_dataloader_from_config(config=val_data_layer_config)
+        self._validation_dl = self.__setup_dataloader_from_config(config=val_data_layer_config)
 
     def setup_test_data(self, test_data_layer_params: Optional[Union[DictConfig, Dict]]):
         if 'shuffle' not in test_data_layer_params:
             test_data_layer_params['shuffle'] = False
-        self.__test_dl = self.__setup_dataloader_from_config(config=test_data_layer_params)
+        self._test_dl = self.__setup_dataloader_from_config(config=test_data_layer_params)
 
     def setup_optimization(self, optim_config: Optional[Union[DictConfig, dict]] = None) -> torch.optim.Optimizer:
-        self.__optimizer = super().setup_optimization(optim_config)
-        self.__scheduler = prepare_lr_scheduler(
-            optimizer=self.__optimizer, scheduler_config=optim_config, train_dataloader=self.__train_dl
+        # Setup optimizer and scheduler
+        if 'sched' in optim_config and 'trainer' in optim_config:
+            if optim_config.trainer.max_steps is None:
+                if optim_config.trainer.gpus == 0:
+                    # training on CPU
+                    iters_per_batch = optim_config.trainer.max_epochs / float(
+                        optim_config.trainer.num_nodes * optim_config.trainer.accumulate_grad_batches
+                    )
```

Comment:
```python
iters_per_batch = self._cfg.model.pl.trainer.max_epochs / float(
                        self._cfg.model.pl.trainer.num_nodes * self._cfg.model.pl.trainer.accumulate_grad_batches
                    )

```

Reply:
but without . model since self._cfg is already model's config